tomviz/python/TV_Filter.py

```diff
@@ -0,0 +1,121 @@
+from tomviz import utils
+import numpy as np
+from numpy.fft import fftn, fftshift, ifftn, ifftshift
+import tomviz.operators
+
+
+class ArtifactsTVOperator(tomviz.operators.CancelableOperator):
+
+    def transform_scalars(self, dataset, Niter=100, a=0.1,
+                          wedgeSize=5, kmin=5, theta=0):
+        """
+        Remove Structured Artifacts with Total Variation Minimization"""
+
+        #Import information from dataset
+        array = utils.get_array(dataset)
+        (nx, ny, nz) = array.shape
+
+        # Convert angle from Degrees to Radians.
+        theta = (theta+90)*(np.pi/180)
+        dtheta = wedgeSize*(np.pi/180)
+
+        #Create coordinate grid in polar
+        x = np.arange(-nx/2, nx/2-1, dtype=np.float64)
+        y = np.arange(-ny/2, ny/2-1, dtype=np.float64)
+        [x, y] = np.meshgrid(x, y, indexing='ij')
+        rr = (np.square(x) + np.square(y))
+        phi = np.arctan2(x, y)
+
+        #Create the Angular Mask
+        mask = np.ones((nx, ny), dtype=np.int8)
+        mask[np.where((phi >= (theta-dtheta/2)) &
+                      (phi <= (theta+dtheta/2)))] = 0
+        mask[np.where((phi >= (np.pi+theta-dtheta/2)) &
+                      (phi <= (np.pi+theta+dtheta/2)))] = 0
+        mask[np.where((phi >= (-np.pi+theta-dtheta/2)) &
+                      (phi <= (-np.pi+theta+dtheta/2)))] = 0
+        mask[np.where(rr < np.square(kmin))] = 1 # Keep values below rmin.
+        mask = np.array(mask, dtype=bool)
+
+        # Initialize Progress bar.
+        if nz == 1:
```

Comment:
Instead of having two different logic for the progress depending on how many images are in the stack, I think it could be generalized.
Something like `self.progress.maximum = nx * Niter`
And then in the inner loop over `Niter` do `self.progress.value = i * Niter + j `
And you can also get rid of manually setting the `step` variable